src/Microsoft.Health.Dicom.Functions.Client/Extensions/AzureComponentFactoryExtensions.cs

```diff
@@ -12,13 +17,76 @@ namespace Microsoft.Health.Dicom.Functions.Client.Extensions;
 
 internal static class AzureComponentFactoryExtensions
 {
-    public static TClient CreateClient<TOptions, TClient>(this AzureComponentFactory factory, IConfigurationSection configuration)
+    public static BlobServiceClient CreateBlobServiceClient(this AzureComponentFactory factory, IConfigurationSection configuration)
```

Comment:
Un-related to this. I just reviewed the last commit on refactoring. 

Can you help me understand you comment and adding more testing of components?

I tried to reduce the amount of work we do in HealthCheck to improve performance and looks like the last commit is doing more now.

Also, looks like we are catching regressiong too late. Can you add MI tests here.